src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java

```diff
@@ -30,81 +38,108 @@ public PostgresSQLConnection(DbConnection conn) {
 
   @Override
   public SQLConnection setAutoCommit(boolean autoCommit, Handler<AsyncResult<Void>> handler) {
-    throw new UnsupportedOperationException();
+    inAutoCommit.set(autoCommit);
```

Comment:
This should be only set if the rest of the code succeeded, in other words, right before any: `handler.handle(Future.succeededFuture())`